### PR TITLE
Add table oci_database_pluggable_database. Closes #290

### LINKED ISSUE #290: Add table oci_database_pluggable_database
**References**
https://registry.terraform.io/providers/oracle/oci/latest/docs/resources/database_pluggable_database
https://docs.oracle.com/en-us/iaas/Content/Database/Tasks/pluggable_database_operations.htm

### PR DESCRIPTION
**Pull request by karanpopat**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
TF resource creation for the parent resource DB system has issues, also CLI is not feasible.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select
  pdb_name,
  id,
  lifecycle_state,
  time_created
from
  oci_database_pluggable_database
+-------------+------------------------------------------------------------------------------------------------------+-----------------+---------------------------+
| pdb_name    | id                                                                                                   | lifecycle_state | time_created              |
+-------------+------------------------------------------------------------------------------------------------------+-----------------+---------------------------+
| DB0324_pdb1 | ocid1.pluggabledatabase.oc1.ap-mumbai-1.anrg6ljr6igdexaagk5z4f32sheetvpgzrutdku2vmtl72kpcj3xqzes6wxq | TERMINATED      | 2022-03-24T12:43:32+05:30 |
+-------------+------------------------------------------------------------------------------------------------------+-----------------+---------------------------+

> select
  pdb_name,
  id,
  lifecycle_state,
  time_created
from
  oci_database_pluggable_database
where
  lifecycle_state <> 'AVAILABLE'
+-------------+------------------------------------------------------------------------------------------------------+-----------------+---------------------------+
| pdb_name    | id                                                                                                   | lifecycle_state | time_created              |
+-------------+------------------------------------------------------------------------------------------------------+-----------------+---------------------------+
| DB0324_pdb1 | ocid1.pluggabledatabase.oc1.ap-mumbai-1.anrg6ljr6igdexaagk5z4f32sheetvpgzrutdku2vmtl72kpcj3xqzes6wxq | TERMINATED      | 2022-03-24T12:43:32+05:30 |
+-------------+------------------------------------------------------------------------------------------------------+-----------------+---------------------------+
```
</details>

### CODE REVIEW

File: docs/tables/oci_database_pluggable_database.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+select
+  pdb_name,
+  id,
+  lifecycle_state,
+  time_created
+from
+  oci_database_pluggable_database;
+```
+
+### List pluggable databases that are not available
```
Comment (ParthaI) [inline docs/tables/oci_database_pluggable_database.md:19]:
```suggestion
### List unavailable pluggable databases
```

File: oci/table_oci_database_pluggable_database.go [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+
+	// Check for additional filters
+	if equalQuals["pdb_name"] != nil {
+		dbName := equalQuals["pdb_name"].GetStringValue()
+		request.PdbName = types.String(dbName)
+	}
+
+	if equalQuals["lifecycle_state"] != nil {
+		lifecycleState := equalQuals["lifecycle_state"].GetStringValue()
+		if isValidState(lifecycleState) {
```
Comment (ParthaI) [inline oci/table_oci_database_pluggable_database.go:179]: Avoid making API call if the lifecycle state is not valid.

**Review by ParthaI — COMMENTED** [review]
@karanpopat please see the comments and make changes wherever required
Q1. We don't have integration test for it, terraform doesn't support this? I think terraform support this https://registry.terraform.io/providers/oracle/oci/latest/docs/resources/database_pluggable_database
Q2. can you please update docs with more example queries?

**Review by ParthaI — APPROVED** [review]
LGTM

**Review by bigdatasourav — APPROVED** [review]
LGTM

**Pull request merged**